x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Authenticator.java

```diff
@@ -44,13 +45,6 @@ public interface Authenticator {
     @Nullable
     AuthenticationToken extractCredentials(Context context);
 
-    /**
-     * Whether authentication with anonymous or fallback user is allowed after this authenticator.
-     */
-    default boolean canBeFollowedByNullTokenHandler() {
-        return true;
-    }
-
```

Comment:
This has been removed in favor of the change in from `RealmsAuthenticator#extractCredentials`.
This, I think, was a convoluted way to express that "the null token case must only be handled if no realm was able to extract a token from the request".